### ISSUE TITLE
Make il8n tools available from `python -m`

### ISSUE THREAD

**Issue by 80ec8472-4906-4627-bd2e-14667fa8ed0c** [python/cpython#81018]
BPO | [36837](https://bugs.python.org/issue36837)
--- | :---
Nosy | @warsaw, @abadger, @bbkane

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-05-07.16:23:26.059>
labels = []
title = 'Make il8n tools available from `python -m`'
updated_at = <Date 2019-05-11.15:26:52.551>
user = 'https://github.com/bbkane'
```

bugs.python.org fields:
```python
activity = <Date 2019-05-11.15:26:52.551>
actor = 'a.badger'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = []
creation = <Date 2019-05-07.16:23:26.059>
creator = 'bbkane'
dependencies = []
files = []
hgrepos = []
issue_num = 36837
keywords = []
message_count = 5.0
messages = ['341771', '341876', '341999', '342005', '342198']
nosy_count = 3.0
nosy_names = ['barry', 'a.badger', 'bbkane']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue36837'
versions = []
```

</p></details>

**Comment by tomasr8** [comment]
Yeah I was going to re-propose this once I have fixed most of the issues w/ pygettext and msgfmt. I've spent quite a long time on just fixing pygettext (didn't even get to msgfmt yet 😆 ) and I had this in mind as the final step once the tools are ready to be exposed for a wider use (currently they're somewhat hidden inside Tools which is not even available on most distributions IIRC).

If you don't mind, I'd like to do this part myself. However, there's still a couple of things I'd like to implement in pygettext and msgfmt. If you'd like to help I'm happy to split them with you 🙂

**Comment by StanFromIreland** [comment]
Sure what are you planning?

**Comment by tomasr8** [comment]
Couple things :)

- Deprecate or remove the  `extractall` option (currently a noop)
- Possibly add `--omit-header` (would also be useful for some our tests as it would allows us to switch to a full POT in `lib/test/translationdata`)
- `width` not being respected for msgids but only for location comments
- Some weird formatting for strings containing format specifiers
- Cleaning up some oboslete tests from `test_i18n`
- and a couple more things which I still need to look into

Let me know which one you'd like to take and I can explain in more detail

**Comment by StanFromIreland** [comment]
I'll look through the 35 tests in test_i18n.

**Comment by StanFromIreland** [comment]
I was unable to find any tests I would consider obsolete, maybe I missed something? There is some duplication however. `test_funcdocstring_multiple_funcs` covers three of the previous cases.

**Comment by StanFromIreland** [comment]
I can remove `extractall`.
